scripts/lib/CIME/test_scheduler.py

```diff
@@ -606,8 +606,9 @@ def _xml_phase(self, test):
         config_test = Tests()
         testnode = config_test.get_test_node(test_case)
         envtest.add_test(testnode)
-        # Determine the test_case from the test name
-        test_case, case_opts = parse_test_name(test)[:2]
+
+        if compiler == 'nag':
+            envtest.set_value("FORCE_BUILD_SMP","FALSE")
```

Comment:
I'm wondering if there's a different way to fix this issue, rather than putting this compiler-specific logic in the code. I spent some time looking into where `FORCE_BUILD_SMP` is set to TRUE. My guess, based on this, is that you're seeing problems in ERP tests. Is that right? If so, I'd prefer to see a fix that gets at the root of the problem. Specifically: it looks like `FORCE_BUILD_SMP` is *always* set to TRUE for ERP tests. But it feels like that shouldn't be done: it seems like this only needs to be set if the ERP test is set up with multi-threading, via something like this (untested!):

```diff
diff --git a/config/config_tests.xml b/config/config_tests.xml
index 709fc35d3..51c7b3f2e 100644
--- a/config/config_tests.xml
+++ b/config/config_tests.xml
@@ -256,7 +256,6 @@ NODEFAIL          Tests restart upon detected node failure. Generates fake failu
     <STOP_OPTION>ndays</STOP_OPTION>
     <STOP_N>11</STOP_N>
     <DOUT_S>FALSE</DOUT_S>
-    <FORCE_BUILD_SMP>TRUE</FORCE_BUILD_SMP>
     <REST_N>$STOP_N / 2 + 1 </REST_N>
     <REST_OPTION>$STOP_OPTION</REST_OPTION>
     <HIST_OPTION>$STOP_OPTION</HIST_OPTION>
diff --git a/scripts/lib/CIME/SystemTests/erp.py b/scripts/lib/CIME/SystemTests/erp.py
index f80c82ce7..ef514e28e 100644
--- a/scripts/lib/CIME/SystemTests/erp.py
+++ b/scripts/lib/CIME/SystemTests/erp.py
@@ -27,15 +27,21 @@ def __init__(self, case):
 
     def _case_two_setup(self):
         # halve the number of tasks and threads
+        threading_in_case_one = False
         for comp in self._case.get_values("COMP_CLASSES"):
             ntasks    = self._case1.get_value("NTASKS_{}".format(comp))
             nthreads  = self._case1.get_value("NTHRDS_{}".format(comp))
             rootpe    = self._case1.get_value("ROOTPE_{}".format(comp))
             if ( nthreads > 1 ):
                 self._case.set_value("NTHRDS_{}".format(comp), int(nthreads/2))
+                threading_in_case_one = True
             if ( ntasks > 1 ):
                 self._case.set_value("NTASKS_{}".format(comp), int(ntasks/2))
                 self._case.set_value("ROOTPE_{}".format(comp), int(rootpe/2))
+        if threading_in_case_one:
+            # If case one is built threaded, then force case two to be built threaded as
+            # well, to ensure it is bit-for-bit
+            self._case.set_value("FORCE_BUILD_SMP", "TRUE")
 
         RestartTest._case_two_setup(self)
         self._case.case_setup(test_mode=True, reset=True)
```

@jedwards4b @jgfouca does this make sense to both of you?

Reply:
Works for me, applied.

Reply:
Thanks. Is it possible that this will change answers for non-threaded ERP tests on some compilers? If so, we should change the test status line in the PR comment.

Reply:
My vote is to keep ERP with threading enabled by default (FORCE_BUILD_SMP=TRUE). Threading was always ON in ERP tests. ERP=ERS+PEM+PET. Most default PE-layouts are single-threaded. Disabling threading in ERP can miss out on catching a large number of bugs.